### PR TITLE
create lounge map component

### PR DESCRIPTION
**Pull request by Jaden51**
map

The main purpose of this PR is to create the lounge map to display the current state of the lounge. Whether a PC is occupied, available or the time is up. To accomplish this:
- Created a new API endpoint to collect all the current active PC's.
- Created a map and a pc-station components to display the info.
- Created another zustand store to store the global map state

## Notes
- No styling done according to the figma
- No check out pc station done

image

### CODE REVIEW

**Comment by joshhwuu** [comment]
copilot is a crazy commit name

**Review by vickiszhang — APPROVED** [review]
🤠

File: client/src/app/check-in/check-in.tsx [hunk 1]
```diff
@@ -34,6 +34,7 @@ const CheckIn = () => {
       membershipTier: addedProfile.membership_tier,
       notes: addedProfile.notes,
     });
+    fetchPCStatus();
```
Comment (joshhwuu) [inline client/src/app/check-in/check-in.tsx:37]: Fabulous!

**Review by joshhwuu — APPROVED** [review]
awesome work, keep it up!

**Pull request merged**